### ISSUE TITLE
Surprisingly good for summarisation

### ISSUE THREAD

**Issue by aria-blackford** [SkBlaz/rakun#10]
Just wanted to make sure you were aware of how well rakun2 can perform as part of an extractive summarisation pipeline. Is this something you'd explored?

**Comment by SkBlaz** [comment]
Hi @aria-blackford , we never evaluated it for this task, cool to hear. You can see some of the works that cite this for down-stream use casess. Lmk if you work on this, would be interesting to see/hear about results there.

edit: forgot to share this PoC based on rakun https://summarizer.ijs.si/, this shows summarization capability

**Comment by aria-blackford** [comment]
I just realised that I should have posted in rakun2, since that's what I'm using. Sorry!

Is the code behind https://summarizer.ijs.si/ available?

I'm just ranking sentences based on their inclusion of rakun2 generated keyphrases. Very simple.

I also tried using biased textrank with rakun2 keyphases influencing the bias.

**Comment by SkBlaz** [comment]
No problem - I'll add code there soon, thanks for the interest

Jonathon ***@***.***> schrieb am Di., 28. Jan. 2025, 07:32:

**Comment by SkBlaz** [comment]
@aria-blackford I've added an example of e2e summarization as https://github.com/SkBlaz/rakun2/blob/main/examples/summarize_text.py, feel free to adapt, experiment with/modify. This is the base for the online tool linked above.
